overlord/devicestate/handlers.go

```diff
@@ -19,18 +19,59 @@
 package devicestate
 
 import (
+	"fmt"
+	"os/exec"
 	"time"
 
 	"gopkg.in/tomb.v2"
 
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/overlord/snapstate"
 	"github.com/snapcore/snapd/overlord/state"
 )
 
+func (m *DeviceManager) doMarkPreseeded(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	snaps, err := snapstate.All(st)
+	if err != nil {
+		return err
+	}
+
+	if m.preseed {
```

Comment:
shouldn't this  start with just:

```
if !m.preseed {
    return nil
}
```

Reply:
This handler gets extra stuff to do related to enabling services if m.preseed is false, and it goes below this if clause in the followup PR #8046. It can be simplified indeed once we fix services.